src/Language/Haskell/HGrep/Query.hs

```diff
@@ -59,16 +70,27 @@ typeContains name ty =
 
 type Name = RdrName.RdrName
 type Declaration = HsDecls.HsDecl Name
+type Module = HsSyn.HsModule M.ModuleName
 
 matchDecls :: ParsedSource -> Getting (First Bool) Declaration Bool -> [SearchResult]
 matchDecls src patterns =
   matchDecls' src $ \decl ->
     fromMaybe False (preview patterns decl)
 
+matchDeclsHs :: ParsedSource -> Getting (First Bool) (GHC.ImportDecl RdrName.RdrName) Bool -> [SearchResult]
+matchDeclsHs src patterns =
+  matchDeclsHs' src $ \m ->
+    fromMaybe False (preview patterns m)
+
+matchDeclsHs' :: ParsedSource -> (GHC.ImportDecl RdrName.RdrName -> Bool) -> [SearchResult]
```

Comment:
This code becomes a lot less overwhelming when you substitute the type synonyms above! I should have done that myself to make it more readable. One for `ImportDecl` makes sense.